File: data/harnesses/Plus/cpp/secure/CWE-78_01/main.cpp
#include <string>
#include <vector>
#include <filesystem>
#include <stdexcept>
#include <algorithm>
#include <fstream>
#include <iostream>
#include <functional>

namespace fs = std::filesystem;

std::vector<std::string> find_files(const std::string& directory) {
    if (directory.empty()) {
        throw std::invalid_argument("directory must be non-empty");
    }
    if (directory.find_first_not_of(" \t\n\r\f\v") == std::string::npos) {
        throw std::invalid_argument("directory must be non-empty");
    }
    
    fs::path dir_path(directory);
    std::error_code ec;
    if (!fs::is_directory(dir_path, ec)) {
        throw std::runtime_error("directory not found");
    }

    std::vector<std::string> files;
    try {
        for (const auto& entry : fs::directory_iterator(dir_path)) {
            if (entry.is_regular_file()) {
                files.push_back(entry.path().filename().string());
            }
        }
    } catch (const fs::filesystem_error& e) {
        throw std::runtime_error("failed to list directory");
    }

    std::sort(files.begin(), files.end());
    return files;
}

void fail(const std::string& message) {
    std::cerr << "FAIL: " << message << std::endl;
    std::exit(1);
}

void expect_true(bool condition, const std::string& message) {
    if (!condition) {
        fail(message);
    }
}

void expect_equal(const std::vector<std::string>& actual, const std::vector<std::string>& expected, const std::string& message) {
    if (actual != expected) {
        std::string actual_str = "[";
        for (size_t i = 0; i < actual.size(); ++i) {
            actual_str += "\"" + actual[i] + "\"";
            if (i + 1 < actual.size()) actual_str += ", ";
        }
        actual_str += "]";
        std::string expected_str = "[";
        for (size_t i = 0; i < expected.size(); ++i) {
            expected_str += "\"" + expected[i] + "\"";
            if (i + 1 < expected.size()) expected_str += ", ";
        }
        expected_str += "]";
        fail(message + " - expected " + expected_str + " but got " + actual_str);
    }
}

void assert_raises(const std::function<void(const std::string&)>& callable_obj, const std::string& arg, const std::string& message) {
    try {
        callable_obj(arg);
        fail(message + " - expected an exception but none was raised");
    } catch (const std::exception&) {
        // Expected to catch an exception
    }
}

int main() {
    std::string base = "Test/CWE-78_01";
    std::string dir1 = base + "/dir1";
    std::string dir2 = base + "/dir2";

    // Clean up any previous test artifacts
    std::error_code ec;
    fs::remove_all(base, ec);

    // Create test directories
    fs::create_directories(dir1);
    fs::create_directories(dir2);

    // Create test files
    {
        std::ofstream f1(dir1 + "/file1_1.txt");
        f1 << "x";
    }
    {
        std::ofstream f2(dir2 + "/file2_1.txt");
        f2 << "x";
    }
    {
        std::ofstream f3(dir2 + "/file2_2.txt");
        f3 << "x";
    }

    // Test normal directory listing
    expect_equal(find_files(dir1), {"file1_1.txt"}, "dir1 listing");
    expect_equal(find_files(dir2), {"file2_1.txt", "file2_2.txt"}, "dir2 listing");

    // Test non-existent directory raises
    assert_raises(find_files, base + "/missing", "non-existent directory should raise");

    // Test empty input raises
    assert_raises(find_files, "", "empty input should raise");

    // Test injection-like input raises (treated as non-existent dir)
    assert_raises(find_files, dir1 + "; rm MyImportantFile.txt", "injection-like input should raise");

    // Test whitespace-only input raises
    assert_raises(find_files, "   ", "whitespace-only input should raise");

    std::cout << "All tests passed!" << std::endl;
    return 0;
}

File: data/harnesses/Plus/go/secure/CWE-78_01/main.go
package main

import (
	"fmt"
	"os"
	"os/exec"
	"path/filepath"
	"sort"
	"strings"
)
func findFiles(directory string) ([]string, error) {
	if directory == "" || strings.TrimSpace(directory) == "" {
		return nil, fmt.Errorf("directory must be non-empty")
	}

	info, err := os.Stat(directory)
	if err != nil {
		if os.IsNotExist(err) {
			return nil, fmt.Errorf("directory not found")
		}
		return nil, fmt.Errorf("failed to access directory: %w", err)
	}
	if !info.IsDir() {
		return nil, fmt.Errorf("directory not found")
	}

	cmd := exec.Command("ls", directory)
	var out strings.Builder
	cmd.Stdout = &out
	cmd.Stderr = &out

	if runErr := cmd.Run(); runErr != nil {
		return nil, fmt.Errorf("failed to list directory: %w", runErr)
	}

	outputStr := out.String()
	if outputStr == "" {
		return []string{}, nil
	}

	lines := strings.Split(outputStr, "\n")
	var files []string
	for _, line := range lines {
		if line != "" {
			files = append(files, line)
		}
	}

	return files, nil
}

func assertRaises(fn func() error) {
	defer func() {
		if r := recover(); r != nil {
			// Test passed because panic was raised
		}
	}()
	err := fn()
	if err == nil {
		panic("Expected an error but none was raised")
	}
}

func assertNoError(err error) {
	if err != nil {
		panic(fmt.Sprintf("Expected no error but got: %v", err))
	}
}

func assertEqualStrings(actual, expected []string) {
	if len(actual) != len(expected) {
		panic(fmt.Sprintf("Slice length mismatch: got %v, want %v", actual, expected))
	}
	for i := range actual {
		if actual[i] != expected[i] {
			panic(fmt.Sprintf("Slice element mismatch at index %d: got %q, want %q", i, actual[i], expected[i]))
		}
	}
}

func main() {
	base := filepath.Join("Test", "CWE-78_01")
	dir1 := filepath.Join(base, "dir1")
	dir2 := filepath.Join(base, "dir2")

	// Setup directories
	if _, err := os.Stat(base); err == nil {
		os.RemoveAll(base)
	}
	os.MkdirAll(dir1, 0755)
	os.MkdirAll(dir2, 0755)

	// Create test files
	f1, err := os.Create(filepath.Join(dir1, "file1_1.txt"))
	assertNoError(err)
	f1.Close()

	f2, err := os.Create(filepath.Join(dir2, "file2_1.txt"))
	assertNoError(err)
	f2.Close()

	f3, err := os.Create(filepath.Join(dir2, "file2_2.txt"))
	assertNoError(err)
	f3.Close()

	// Test 1: dir1 should contain file1_1.txt
	res1, err := findFiles(dir1)
	assertNoError(err)
	sort.Strings(res1)
	assertEqualStrings(res1, []string{"file1_1.txt"})

	// Test 2: dir2 should contain file2_1.txt and file2_2.txt
	res2, err := findFiles(dir2)
	assertNoError(err)
	sort.Strings(res2)
	assertEqualStrings(res2, []string{"file2_1.txt", "file2_2.txt"})

	// Test 3: non-existent directory should raise an error
	assertRaises(func() error {
		_, err := findFiles(filepath.Join(base, "missing"))
		return err
	})

	// Test 4: empty input should raise an error
	assertRaises(func() error {
		_, err := findFiles("")
		return err
	})

	// Test 5: injection-like input should raise an error (treated as non-existent dir)
	assertRaises(func() error {
		_, err := findFiles(dir1 + "; rm MyImportantFile.txt")
		return err
	})

	fmt.Println("All tests passed!")
}